HWIOAuthBundle.php

```diff
@@ -37,11 +37,19 @@ public function build(ContainerBuilder $container)
         /** @var SecurityExtension $extension */
         $extension = $container->getExtension('security');
 
-        // Symfony < 5.4 BC layer
-        if (interface_exists(AuthenticationProviderInterface::class)) {
-            $extension->addSecurityListenerFactory(new OAuthFactory()); // @phpstan-ignore-this-line Symfony < 5.4 BC layer
+        // Symfony < 5.1 BC layer: support new Authenticator-based security system in Symfony 5.1+
+        // and old security system in all Symfony versions.
+        if (interface_exists(AuthenticatorFactoryInterface::class)) {
+            if (method_exists($extension, 'addAuthenticatorFactory')) {
+                $extension->addAuthenticatorFactory(new OAuthAuthenticatorFactory());
+            } else {
+                /*
+                 * @deprecated since Symfony 5.4, use "addAuthenticatorFactory()" instead
+                 */
+                $extension->addSecurityListenerFactory(new OAuthAuthenticatorFactory());
+            }
         } else {
-            $extension->addAuthenticatorFactory(new OAuthAuthenticatorFactory());
+            $extension->addSecurityListenerFactory(new OAuthFactory());
         }
```

Comment:
All this should be reverted IMHO.

Reply:
ohne changes hier was OAuthAuthenticator not registerd under symfony 5.4
s. #1835

Reply:
I have reverted it and waiting for your fix